synapsis/src/main/java/com/soen/synapsis/appuser/registration/RegistrationService.java

```diff
@@ -41,7 +44,20 @@ public String register(RegistrationRequest request) {
                 new AppUser(request.getName(),
                         request.getPassword(),
                         request.getEmail(),
-                        requestedRole)
+                        requestedRole,
+                        AuthProvider.LOCAL)
         );
     }
+
+    public AppUser retrieveUserOrRegisterSSOIfNotExists(String name, String email) {
```

Comment:
method name is kind of confusing, maybe consider either retrieveUser or registerSSOUser and comment the explanation of extra logic (check existing first, etc...)

Reply:
I renamed the method to something simpler and added a comment.